common/src/main/java/com/wynntils/models/items/annotators/gui/CosmeticTierAnnotator.java

```diff
@@ -31,8 +32,11 @@ public ItemAnnotation getAnnotation(ItemStack itemStack, StyledText name) {
     }
 
     private static boolean isCosmetic(ItemStack itemStack) {
-        for (Component c : LoreUtils.getTooltipLines(itemStack)) {
-            if (COSMETIC_PATTERN.matcher(c.getString()).matches()) return true;
+        List<Component> lore = LoreUtils.getTooltipLines(itemStack);
+        for (int i = lore.size() - 1; i >= 0; i--) { // Reverse iteration is slightly faster for most users
```

Comment:
I think this is a premature optimization. It makes the code much harder to read, and I doubt it has any serious effect on performance. If you think so, please make measurements, otherwise revert this part of the fix.